functions.py

```diff
@@ -1,34 +1,86 @@
 import time
+SEC_IN_YEAR = 31536000.0
 
-def ReadLines(nameoffile):
+#function returns data fron text in list(array)
+def readStateFile(nameoffile):
     with open(nameoffile,'r') as f:
-        arrayline = f.readlines()
-        return arrayline
+        arrdata = f.readlines()
+        if arrdata != None:
+            return arrdata
+        else:
+            print("Error, empty file, try another one")
+            return 0
+        
+#check the order by date        
+def checkOrderByDate(arrdata):
+    previousdate = 0.0
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if float(arrline[1]) < previousdate:
+            print("array doesnt ordered by date")
+            print("previousdate:", previousdate, "date:", arrline[1]) 
+            return 1
+        previousdate = float(arrline[1])
+    return 0
+
+# sorting by time btw
+def SortByDate(strings):
+    sorted_strings = sorted(strings, key=lambda line: float(line.split(",")[1]))
+    return sorted_strings
 
 #how much is not taxed
-def SellWithoutTax():
-    arrayline = ReadLines('new_register.txt')
+def SellWithoutTax(arrdata):
     sum = 0.0
-    i = 0
-    arr = arrayline[i].split(',', 2)
-    while (time.time() - float(arr[1])) >= 31536000.0:
-        sum += float(arr[0])
-        i += 1
-        arr = arrayline[i].split(',', 2)
+    for i in range(0, len(arrdata)):
+        arrline  = arrdata[i].split(',', 2)
+        if (time.time() - float(arrline[1])) >= SEC_IN_YEAR:
+            sum += float(arrline[0])
     print("can sell without tax:", sum)
-    return sum, i
+    return sum
 
-#benefit from not taxed
-def Benefit(sum, firsttax,price):
-    arrayline = ReadLines('new_register.txt')
-    totalprice = 0.0
-    for i in range(0, firsttax):
-        arr = arrayline[i].split(',', 2)
-        totalprice += float(arr[0]) * float(arr[2])
-    print("had payed:", totalprice)
-    print("will get:", sum * price)    
-    print("benefit:", sum*price - totalprice)
-    return (sum*price - totalprice)
+#counting benefit 
+def Benefit(sum, price, arrdata):

```

Comment:
May be a bit more details on what this function does ? 

Basically, I might be interested in something like 
[benefit, taxable benefit] output.